src/main/java/picard/analysis/CollectMultipleMetrics.java

```diff
@@ -557,11 +575,55 @@ public int doWork() {
                     REF_FLAT,
                     IGNORE_SEQUENCE,
                     INCLUDE_UNPAIRED);
+
+            if (additionalArguments.containsKey(program)) {
+                final CommandLineParser commandLineParser = getCommandLineParser(instance);
+                final boolean success = commandLineParser.parseArguments(System.err, additionalArguments.get(program).toArray(new String[0]));
+                if (!success) {
+                    throw new IllegalArgumentException("Failed to parse arguments for " + program);
+                }
+                additionalArguments.remove(program);
+            }
+
             instance.setDefaultHeaders(getDefaultHeaders());
             programs.add(instance);
         }
+        if(!additionalArguments.isEmpty()){
+            throw new IllegalArgumentException("EXTRA_ARGUMENT values were provided, but appropriate program wasn't requested:" +
+                    additionalArguments.entrySet().stream().map(e->e.getKey().toString()+"::"+e.getValue().toString()).collect(Collectors.joining()));
+        }
         SinglePassSamProgram.makeItSo(INPUT, REFERENCE_SEQUENCE, ASSUME_SORTED, STOP_AFTER, programs);
 
         return 0;
     }
+
+    private static Map<ProgramInterface, List<String>> processAdditionalArguments(final List<String> arguments) {
+        final Map<ProgramInterface, List<String>> map = new HashMap<>();
+        final Pattern pattern = Pattern.compile("(?<program>.*)::(?<argumentAndValue>.+?)( +(?<optionalValue>.+))?");
+        for (String str : arguments) {
+            final Matcher matcher = pattern.matcher(str);
+            if (!matcher.matches()) {
+                throw new IllegalArgumentException("couldn't understand EXTRA_ARGUMENT " + str + " it doesn't conform to the regular expression given by " + pattern.pattern());
+            }
+            final String programName = matcher.group("program");
+            final ProgramInterface program;
+            try {
+                program = Program.valueOf(programName);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Couldn't find program with value " + programName, e);
+            }
+            final String argumentAndValue = matcher.group("argumentAndValue");
+
+
+            map.computeIfAbsent(program, (k)->new ArrayList<>()).add(argumentAndValue);
+
+            final String optionalValue = matcher.group("optionalValue");
+            if (optionalValue!=null && !optionalValue.equals("")) {
+                map.get(program).add(optionalValue);
+            }
```

Comment:
This code path needs a test case, preferably one where another  arg appears in the EXTRA_ARGS list after the optional one.

Reply:
there is a test case...when there's a space after the "=" in 
```
      "EXTRA_ARGUMENT=CollectInsertSizeMetrics::HISTOGRAM_WIDTH= 58",
          
```

Reply:
@yfarjoun I guess I misunderstood what the purpose of this code path is then. I interpreted "optional" to mean that this value was truly optional, but this is just to tolerate the space ? Although the test case passes as written (in that 58 gets propagated), this syntax won't work from the command line this way (without quoting at least - it does work with quoting).

But more importantly, although the tests seem to pass with both parsers (because args is already parsed in to an array of strings ?), I've tried a few variations manually and I haven't yet succeeded in passing extra args using the Barclay parser (-Dpicard.useLegacyParser=false) from the command line at all. What is the right/intended syntax to use for extra args in that case ?

Reply:
you'd need to quote the argument and the value together:

```
java -Dpicard.useLegacyParser=false -jar build/libs/picard.jar CollectMultipleMetrics \
-INPUT /Users/farjoun/Picard-public/testdata/picard/sam/summary_alignment_stats_test.sam \
-OUTPUT /var/folders/tc/hy9lszxd1dg9cf4bky51mrrd9k3s6g/T/alignmentMetrics6690349237641187401 \
-REFERENCE_SEQUENCE /Users/farjoun/Picard-public/testdata/picard/sam/summary_alignment_stats_test.fasta \
-METRIC_ACCUMULATION_LEVEL ALL_READS \
-PROGRAM null \
-PROGRAM CollectAlignmentSummaryMetrics \
-PROGRAM CollectInsertSizeMetrics \
-EXTRA_ARGUMENT "CollectInsertSizeMetrics::--HISTOGRAM_WIDTH 58" \
-EXTRA_ARGUMENT "CollectInsertSizeMetrics::--METRIC_ACCUMULATION_LEVEL LIBRARY" \
-EXTRA_ARGUMENT "CollectInsertSizeMetrics::--METRIC_ACCUMULATION_LEVEL READ_GROUP"
```

Reply:
(I'll update the docs)